source/MaterialXView/ShaderValidators/Glsl/GlslValidator.h

```diff
@@ -0,0 +1,326 @@
+#ifndef MATERIALX_GLSLVALIDATOR_H
+#define MATERIALX_GLSLVALIDATOR_H
+
+#include <MaterialXShaderGen/HwShader.h>
+#include <MaterialXView/ImageHandler.h>
+#include <vector>
+#include <string>
+#include <unordered_map>
+
+namespace MaterialX
+{
+
+/// Error string list type
+using ShaderValidationErrorList = std::vector<std::string>;
+
+/// @class @GlslValidator
+/// Helper class to perform validation of GLSL source code generated by the GLSL code generator.
+///
+/// There are two main interfaces which can be used. One which takes in a HwShader and one which
+/// allows for explicit setting of shader stage code.
+///
+/// The main services provided are:
+///     - Validation: All shader stages are compiled and atteched to a GLSL shader program.
+///     - Introspection: The compiled shader program is examined for uniforms and attributes.
+///     - Binding: Uniforms and attributes which match the predefined variables generated the GLSL code generator
+///       will have values assigned to this. This includes matrices, attribute streams, and textures.
+///     - Rendering: The program with bound inputs will be used to drawing geometry to an offscreen buffer.
+///     An interface is provided to save this offscreen buffer to disk using an externally defined image handler.
+///
+class GlslValidator
+{
+  public:
+    /// Structure to hold information about program inputs
+    struct ProgramInput
```

Comment:
When introspection of actual shader program a list for uniforms and attributes is filled in. This is used as the basis for binding inputs. Some additional HwShader input logic is to be added (e.g. binding using file name provided by HwShader) but for the most part this can all with just the shader stage fragments.